packages/valory/skills/termination_abci/tests/test_rounds.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021-2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Test the rounds of the skill."""
+
+from copy import deepcopy
+from typing import FrozenSet, cast
+
+from packages.valory.skills.abstract_round_abci.base import AbciAppDB, ConsensusParams
+from packages.valory.skills.termination_abci.payloads import BackgroundPayload
+from packages.valory.skills.termination_abci.rounds import (
+    BackgroundRound,
+    Event,
+    SynchronizedData,
+    TerminationRound,
+)
+
+
+MAX_PARTICIPANTS: int = 4
+
+
+def get_participants() -> FrozenSet[str]:
+    """Participants"""
+    return frozenset([f"agent_{i}" for i in range(MAX_PARTICIPANTS)])
+
+
+class BaseRoundTestClass:  # pylint: disable=too-few-public-methods
+    """Base test class for Rounds."""
+
+    synchronized_data: SynchronizedData
+    consensus_params: ConsensusParams
+    participants: FrozenSet[str]
+
+    @classmethod
+    def setup(
+        cls,
+    ) -> None:
+        """Setup the test class."""
+
+        cls.participants = get_participants()
+        cls.synchronized_data = SynchronizedData(
+            AbciAppDB(
+                setup_data=dict(
+                    participants=[cls.participants], all_participants=[cls.participants]
+                ),
+            )
+        )
+        cls.consensus_params = ConsensusParams(max_participants=MAX_PARTICIPANTS)
+
+
+class TestBackgroundRound(BaseRoundTestClass):
+    """Tests for BackgroundRound."""
+
+    def test_run(
+        self,
+    ) -> None:
+        """Run tests."""
+
+        test_round = BackgroundRound(
+            synchronized_data=deepcopy(self.synchronized_data),
```

Comment:
So this test checks whether `synchronized_data` is written to correctly. If I don't copy here, `synchronized_data` will be changed by the round, and the check below will become meaningless because `expected_state` and `actual_state` point to the same instance of `synchronized_data`.
```
        assert (
            cast(SynchronizedData, actual_state).termination_majority_reached
            == cast(SynchronizedData, expected_state).termination_majority_reached
        )
```

Reply:
that is not good and I expect this is the case in other tests then as well. There are 9 such occurrences in transaction settlement test_rounds.py alone, but given the convoluted implementation using generators I am not certain this is in fact an issue. Nonetheless seems worthy of investigation to me, what do you think @Adamantios ?

https://github.com/valory-xyz/open-autonomy/blob/20551f16131c8de0144729f2422bcca91e2210b6/packages/valory/skills/transaction_settlement_abci/tests/test_rounds.py#L244